src/app/shared/models/shelf-configuration.ts

```diff
@@ -1,12 +1,16 @@
 import { Tv2ActionContentType } from './tv2-action'
 
-export interface ShelfConfiguration {
+export interface ShelfConfiguration<T> {
   id: string
-  actionPanelConfigurations: ShelfActionPanelConfiguration[]
+  actionPanelConfigurations: T[]
 }
 
 export interface ShelfActionPanelConfiguration {
   name: string
   rank: number
   actionFilter: Tv2ActionContentType[]
 }
+
+export interface ShelfActionPanelConfigurationWithId extends ShelfActionPanelConfiguration {
+  id: string
+}
```

Comment:
I didn't find any other way to save the selected components after reordering the table if I work with table index. I will check below what your suggestion is.

Reply:
In `action-panel-list.component.ts` you have the variable `selectedActionPanelIds: Set<string>`. 
This could simply be: `selectedActionPanels: Set<ShelfActionPanelConfiguration>` which makes the id redundant.

You need to get out of the mindset of sending the least amount of data around as possible. We are writing object-oriented programming, so we should utilize/take advantage of that and send the objects around.

But, my main point here _isn't_ that you added the id. The id might very well be needed/a good idea to add.

My main point here is that you created a separate interface _and_ made a previous interface generic _just to add a single attribute_.
This is what the code should look like after adding the id:
```
export interface ShelfConfiguration {
  id: string
  actionPanelConfigurations: ShelfActionPanelConfiguration[]
}

export interface ShelfActionPanelConfiguration {
  id: string // <-- This is all you had to add
  name: string
  rank: number
  actionFilter: Tv2ActionContentType[]
}
```